### ISSUE TITLE
epic: Multiple shard support in TMC

### ISSUE THREAD

**Issue by davidfestal** [kcp-dev/kcp#2831]
### TL;DR

Implement the support of multiple shards in the TMC components

### Progress tracking <a href="#progress-tracking" id="progress-tracking">#</a>

EPIC detailed issues and overall progress can be tracked with the following [project view](https://github.com/orgs/kcp-dev/projects/1/views/49)

### Stories

- [ ] Implement the support for several SyncTarget virtual workspace URLs (one per shard) in the Syncer
- [ ] Update all the TMC e2e test fixture to make them support multi-shard
- [ ] Update TMC e2e tests to have them supporting mlti-shard
- [ ] Think about what it means for a coordination controller to be shard-aware
